src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobScheduler.java

```diff
@@ -0,0 +1,100 @@
+     * @param redisPrefixKey the prefix key for the job system. All keys the job system uses will have the prefix redisPrefixKey
+     */
+    public DedupJobScheduler(RDBI rdbi, String redisPrefixKey) {
+        super(rdbi, redisPrefixKey);
+    }
+
+    @Override
+    public boolean schedule(final String tube, final String jobStr, final int runInMillis) {
+
+        try (Handle handle = rdbi.open()) {
+            return 1 == handle.attach(DedupJobSchedulerDAO.class).scheduleJob(
+                    getReadyQueue(tube),
+                    getPaused(tube),
+                    jobStr,
+                    Instant.now().getMillis() + runInMillis);
+        }
+    }
+
+    @Override
+    public List<TimeJobInfo> reserveMulti(final String tube, final long ttrInMillis, final int maxNumberOfJobs) {
+        try (Handle handle = rdbi.open()) {
+            return handle.attach(DedupJobSchedulerDAO.class).reserveJobs(
+                    getReadyQueue(tube),
+                    getRunningQueue(tube),
+                    getPaused(tube),
+                    maxNumberOfJobs,
+                    Instant.now().getMillis(),
+                    Instant.now().getMillis() + ttrInMillis);
+        }
+    }
+
+    @Override
+    public boolean deleteJob(final String tube, String jobStr) {
+        try (Handle handle = rdbi.open()) {
+            return 1 == handle.attach(DedupJobSchedulerDAO.class)
+                              .deleteJob(getReadyQueue(tube), getRunningQueue(tube), jobStr);
+        }
+    }
+
+    public boolean ackJob(final String tube, String jobStr) {
+        try (Handle handle = rdbi.open()) {
+            return 1 == handle.attach(DedupJobSchedulerDAO.class)
+                              .ackJob(getRunningQueue(tube), jobStr);
+        }
+    }
+
+    public List<TimeJobInfo> removeExpiredRunningJobs(String tube) {
+        try (Handle handle = rdbi.open()) {
+            return handle.attach(DedupJobSchedulerDAO.class)
+                         .removeExpiredJobs(getRunningQueue(tube), Instant.now().getMillis());
+        }
+    }
+
+    /**
+     * A job that has been sitting in the ready queue for the last "expirationPeriodInMillis"
+     * is considered expired. Expired jobs are removed from the scheduler and returned to the client.
+     */
+    public List<TimeJobInfo> removeExpiredReadyJobs(String tube, long expirationPeriodInMillis) {
+        try (Handle handle = rdbi.open()) {
+            return handle.attach(DedupJobSchedulerDAO.class)
+                         .removeExpiredJobs(getReadyQueue(tube), Instant.now().minus(expirationPeriodInMillis).getMillis());
+        }
+    }
+
+    public boolean inReadyQueue(String tube, String jobStr) {
+        try (Handle handle = rdbi.open()) {
+            return 1 == handle.attach(DedupJobSchedulerDAO.class)
+                         .inQueue(getReadyQueue(tube), jobStr);
+        }
+    }
+
+    public boolean inRunningQueue(String tube, String jobStr) {
```

Comment:
don't think we need this but added it just to be complete. also useful in testing